### PR TITLE
added patch article functionality

### PR DESCRIPTION
**Pull request by madDwain**
ticket 8

### CODE REVIEW

File: __tests__/test.js [hunk 1]
```diff
@@ -92,6 +92,44 @@ describe("/api/articles/:article_id", () => {
...
+        });
+    });
+    test('should return 404 and a message of article id not found if artcile id not found', () => {
+        return request(app)
+        .patch('/api/articles/9999')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('article id not found')
+        })
+    })
```
Comment (nc-jodie-cheung) [inline __tests__/test.js:131]: Your tests should be:
- Status 400: E.g. PATCH /api/articles/notanumber
- Status 404: E.g. PATCH /api/articles/9999
- Status 400 - E.g. inc_votes value is not a number.

You're not sending anything for your sad path tests, have a look at the notes for post, this will give you an idea of how to set it up: https://notes.northcoders.com/courses/js-back-end/supertest#supertest-setup

File: models/articles-models.js [hunk 2]
```diff
@@ -30,7 +30,22 @@ async function fetchAllArticles() {
...
+    .query(`UPDATE articles SET votes = votes + $2 WHERE article_id = $1 RETURNING *;`, [article_id, inc_votes])
+    .then(({ rows }) => {
+        if (rows.length === 0) {
+            return Promise.reject({status: 404, msg: 'article id not found'})
+        }
+        return rows[0]
+    })
+    .catch((err) => {
+        return Promise.reject(err)
+    })
```
Comment (nc-jodie-cheung) [inline models/articles-models.js:44]: This catch block is redundant and is essentially catching the error that is already occurring in the Promise.reject above. You can remove this 👍

File: __tests__/test.js [hunk 3]
```diff
@@ -92,6 +92,44 @@ describe("/api/articles/:article_id", () => {
...
+      const expectedArticle = {
+        title: "Living in the shadow of a great man",
+        topic: "mitch",
+        author: "butter_bridge",
+        body: "I find this existence challenging",
+        created_at: "2020-07-09T20:11:00.000Z",
+        votes: 110,
+        article_img_url:
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+      };
```
Comment (nc-jodie-cheung) [inline __tests__/test.js:106]: I'd consider adding an object representing the data that is being sent in the PATCH request, for example:

**const expectedArticle = {inc_votes: 10}**

Rather than the whole article, this keeps it concise and you can check the correct amount of votes comes back 😄

File: __tests__/test.js [hunk 4]
```diff
@@ -92,6 +92,44 @@ describe("/api/articles/:article_id", () => {
...
+      return request(app)
+        .patch("/api/articles/1")
+        .send({ inc_votes: 10 })
+        .expect(200)
+        .then(({ body }) => {
+          const article = body;
+          expect(article).toMatchObject(expectedArticle);
+        });
+    });
```
Comment (nc-jodie-cheung) [inline __tests__/test.js:115]: You could also add in a test to check for a negative inc_vote, for example: **{ inc_votes : -100 }**

**Review by nc-jodie-cheung — COMMENTED** [review]
Hey 👋 great start to this ticket, I've left some comments for you to take a look at and refactor. Let me know if you have any questions 😄
